tests/e2e-pw/tests/cart-block/cart-block.merchant.spec.ts

```diff
@@ -0,0 +1,190 @@
+/**
+ * External dependencies
+ */
+import { BlockData } from '@woocommerce/e2e-types';
+import { test, expect } from '@woocommerce/e2e-playwright-utils';
+
+/**
+ * Internal dependencies
+ */
+import { editBlockPage } from '../../utils/navigation/navigation';
+
+const blockData: BlockData = {
+	name: 'Cart',
+	slug: 'woocommerce/cart',
+	mainClass: '.wp-block-woocommerce-cart',
+	selectors: {
+		editor: {
+			block: '.wp-block-woocommerce-cart',
+			insertButton: "//button//span[text()='Cart']",
+		},
+		frontend: {
+			block: '.wp-block-woocommerce-cart',
+		},
+	},
+};
+
+test.describe( 'Merchant → Cart', () => {
+	test.use( { storageState: process.env.ADMINSTATE } );
+
+	test.describe( 'in page editor', () => {
+		test.beforeEach( async ( { page } ) => {
+			await editBlockPage( page, blockData );
+		} );
+
+		test( 'it renders without crashing', async ( { page } ) => {
+			const blockPresence = page.locator(
+				blockData.selectors.editor.block
+			);
+			expect( blockPresence ).toBeTruthy();
+		} );
+
+		test( 'can only be inserted once', async ( { page, editorUtils } ) => {
+			await editorUtils.openGlobalBlockInserter( page );
+			await page.getByPlaceholder( 'Search' ).fill( blockData.slug );
+			const cartBlockButton = await page.locator( 'button', {
+				has: page.locator( `text="${ blockData.name }"` ),
+			} );
+			await expect( cartBlockButton ).toHaveAttribute(
+				'aria-disabled',
+				'true'
+			);
+		} );
+
+		test( 'inner blocks can be added/removed by filters', async ( {
+			page,
+			editor,
+		} ) => {
+			// Begin by removing the block.
+			await editor.selectBlocks( blockData.selectors.editor.block );
+			const options = await page
+				.getByRole( 'toolbar', { name: 'Block tools' } )
+				.getByRole( 'button', { name: 'Options' } );
+			await options.click();
+			const removeButton = await page.getByText( 'Remove Cart' );
+			await removeButton.click();
+			// Expect block to have been removed.
+			await expect(
+				page.locator( blockData.selectors.editor.block )
+			).toHaveCount( 0 );
+
+			// Register a checkout filter to allow `core/table` block in the Checkout block's inner blocks, add
+			// core/audio into the woocommerce/cart-order-summary-block and remove core/paragraph from all Cart inner
+			// blocks.
+			await page.evaluate(
+				"wc.blocksCheckout.registerCheckoutFilters( 'woo-test-namespace'," +
+					'{ additionalCartCheckoutInnerBlockTypes: ( value, extensions, { block } ) => {' +
+					"    value.push('core/table');" +
+					"    if ( block === 'woocommerce/cart-order-summary-block' ) {" +
+					"        value.push( 'core/audio' );" +
+					'    }' +
+					'    return value;' +
+					'}' +
+					'}' +
+					');'
```

Comment:
NIT: If you prefer, you can use the ` to avoid string concatenation:

```suggestion
				`wc.blocksCheckout.registerCheckoutFilters( 'woo-test-namespace', {
	additionalCartCheckoutInnerBlockTypes: ( value, extensions, { block } ) => {
		value.push( 'core/table' );
		if ( block === 'woocommerce/cart-order-summary-block' ) {
			value.push( 'core/audio' );
		}
		return value;
	},
} );`
```